File: sdks/typescript/src/resources/costs.ts
/**
 * Costs resource — cost tracking, history, and budgets.
 *
 * @module resources/costs
 * @see edgequake/crates/edgequake-api/src/handlers/costs.rs
 */

import type {
  BudgetStatus,
  CostHistory,
  CostSummary,
  UpdateBudgetRequest,
} from "../types/costs.js";
import { Resource } from "./base.js";

export class CostsResource extends Resource {
  /** Get cost summary for the workspace. */
  async summary(): Promise<CostSummary> {
    return this._get("/api/v1/costs/summary");
  }

  /** Get cost history over time. */
  async history(): Promise<CostHistory> {
    return this._get("/api/v1/costs/history");
  }

  /** Get current budget status. */
  async budget(): Promise<BudgetStatus> {
    return this._get("/api/v1/costs/budget");
  }

  /** Update budget settings. */
  async updateBudget(request: UpdateBudgetRequest): Promise<BudgetStatus> {
    return this._patch("/api/v1/costs/budget", request);
  }
}

File: sdks/rust/src/resources/costs.rs
//! Costs resource.

use crate::client::EdgeQuakeClient;
use crate::error::Result;
use crate::types::operations::*;

pub struct CostsResource<'a> {
    pub(crate) client: &'a EdgeQuakeClient,
}

impl<'a> CostsResource<'a> {
    /// `GET /api/v1/costs/summary`
    pub async fn summary(&self) -> Result<CostSummary> {
        self.client.get("/api/v1/costs/summary").await
    }

    /// `GET /api/v1/costs/history`
    pub async fn history(&self) -> Result<Vec<CostEntry>> {
        self.client.get("/api/v1/costs/history").await
    }

    /// `GET /api/v1/costs/budget`
    pub async fn budget(&self) -> Result<BudgetInfo> {
        self.client.get("/api/v1/costs/budget").await
    }
}
